uber/models/department.py

```diff
@@ -312,18 +341,35 @@ class Job(MagModel):
         secondary='job_required_role')
     shifts = relationship('Shift', backref='job')
 
-    _repr_attr_names = ['name']
+    __table_args__ = (
+        Index('ix_job_department_id', department_id),
+    )
 
-    # Using a column_property for department_name *might* be more efficient,
-    # due to our prevelant usage of the department_name property.
-    department_name = column_property(
```

Comment:
The column property was definitely not more efficient. I've converted `department_name` and `restricted` to `@hybrid_property` instead.